test/http-client-tests/src/main/java/software/amazon/awssdk/http/SdkHttpClientTestSuite.java

```diff
@@ -113,6 +120,30 @@ public void validatesHttpsCertificateIssuer() throws Exception {
                 .isInstanceOf(SSLHandshakeException.class);
     }
 
+    @Test
+    public void connectionPoolingWorks() throws Exception {
+        int initialOpenedConnections = CONNECTION_COUNTER.openedConnections();
+
+        SdkHttpClientOptions httpClientOptions = new SdkHttpClientOptions();
+        httpClientOptions.trustAll(true);
+        SdkHttpClient client = createSdkHttpClient(httpClientOptions);
+
+        stubForMockRequest(200);
+
+        for (int i = 0; i < 5; i++) {
+            SdkHttpFullRequest req = mockSdkRequest("http://localhost:" + mockServer.port(), SdkHttpMethod.POST);
+            HttpExecuteResponse response =
+                client.prepareRequest(HttpExecuteRequest.builder()
+                                                        .request(req)
+                                                        .contentStreamProvider(req.contentStreamProvider().orElse(null))
+                                                        .build())
+                      .call();
+            response.responseBody().ifPresent(IoUtils::drainInputStream);
+        }
+
+        assertThat(CONNECTION_COUNTER.openedConnections()).isLessThan(initialOpenedConnections + 5);
```

Comment:
Can we be more precise than `lessThan` here? It seems like this should be deterministic. Do we expect exactly `initialOpenedConnections`, `initialOpenedConnections + 1`?

Reply:
I guess the number may depend (and thus vary) upon the client implementation, and how many times we're willing to reuse a connection. Is it fair to say that no client ought to use more than 1 *new* connection here though? Can we do: `isLessThan(initialOpenedConnections + 1)`?

Reply:
In my tests it was fairly consistently `+1` when pooling was working, or `+5` when it was not working. 

Because the test is so indirect (involves cooperation of various HTTP client implementations as well as Wiremock/Jetty, I wasn't confident that hard-coding it to `+1` was going to create a 100% deterministic test, and non-deterministic tests are worse than no test (IMO). I think if it's < 5 we can confidently say that connection pooling is working, but you're right that we don't know to what extent it's working. With the test how it currently is, we could make a change that makes pooling "worse" (whatever that means) without this test catching it, unless it completely breaks pooling altogether.

Let me try running it a few thousand times to see if it ever fails at `+1`.

Reply:
I switched it to +1, it seemed to work just fine. We can reassess if it becomes a problem.